packages/react-dom/src/shared/DOMProperty.js

```diff
@@ -168,23 +149,27 @@ export function shouldSetAttribute(name, value) {
   }
 }
 
-export function getPropertyInfo(name) {
-  return properties.hasOwnProperty(name) ? properties[name] : null;
+// TODO: the fact that we rely on this in many code paths seems bad.
+// It shouldn't be observable whether something is whitelisted if it
+// doesn't have special behavior except being an alias. But at least
+// we're explicit about this now.
+export function isWhitelisted(name: string) {
+  return propertyTypes.has(name);
 }
 
-export function shouldAttributeAcceptBooleanValue(name) {
+export function shouldAttributeAcceptBooleanValue(name: string) {
   if (isReservedProp(name)) {
     return true;
   }
-  let propertyInfo = getPropertyInfo(name);
-  if (propertyInfo) {
-    return (
-      propertyInfo.hasBooleanValue ||
-      propertyInfo.hasStringBooleanValue ||
-      propertyInfo.hasOverloadedBooleanValue
-    );
+  if (propertyTypes.has(name)) {
+    switch (propertyTypes.get(name)) {
```

Comment:
Is the overhead of the switch greater than the double lookup of `has` and `get`?

Reply:
I was trying to avoid making it polymorphic (number/undefined). Maybe that's silly, dunno.